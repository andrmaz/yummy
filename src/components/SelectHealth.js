import React from 'react';
import PropTypes from 'prop-types';
import { Controller } from 'react-hook-form';
import { Flex, FormLabel, Select} from '@chakra-ui/react';

const SelectHealth = ({ control }) => {
    return (
        <Controller
            name="health"
            data-testid='d'
            control={control}
            defaultValue='tree-nut-free'
            render={({ onChange, value }) => 
            <Flex direction='column'>
                <FormLabel textAlign='center' pt='5px' >Health</FormLabel>
                <Select 
                    id='health'
                    onChange={onChange} 
                    value={value} 
                    name='health'
                    size='xs' 
                    w='200px'
                    m='5px'
                    variant="filled"
                    focusBorderColor='blue.500'
                >
                    <option label='tree-nut-free' value='tree-nut-free' />
                    <option label='peanut-free' value='peanut-free' />
                    <option label='vegetarian' value='vegetarian' />
                </Select>
            </Flex>
            }
        />
    );
};

SelectHealth.propTypes = {
    control: PropTypes.object,
};

export default SelectHealth;
