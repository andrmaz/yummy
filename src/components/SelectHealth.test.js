import React from 'react';
import { shallow } from 'enzyme';
import SelectHealth from './SelectHealth';

let wrapper;

describe('SelectHealth Component', () => {
    beforeEach(() => {
        expect.hasAssertions();
    });
    it('should render Component correctly', () => {
        wrapper = shallow(<SelectHealth />);
        expect(wrapper).toMatchSnapshot();
    });
});