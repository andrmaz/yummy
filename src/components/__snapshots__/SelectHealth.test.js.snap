// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectHealth Component should render Component correctly 1`] = `
<Controller
  data-testid="d"
  defaultValue="tree-nut-free"
  name="health"
  render={[Function]}
/>
`;
